management/management.go

```diff
@@ -85,9 +85,17 @@ type Management struct {
 // New creates a new Auth0 Management client by authenticating using the
 // supplied client id and secret.
 func New(domain, clientID, clientSecret string, options ...apiOption) (*Management, error) {
+	if !strings.HasPrefix(domain, "http://") && !strings.HasPrefix(domain, "https://") {
+		domain = "https://" + domain
+	}
+
+	u, err := url.Parse(domain)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is great, so much better! From the playground link you posted I got this idea, what if we check for the presense of `"//"` instead, so we can account for any scheme, and replace it with `https://`.

```
if i := strings.Index(s, "//"); i != -1 {
	s = s[i+2:]
}
s = "https://" + s

u, err := url.Parse(domain)
if err != nil {
	return nil, err
}

if u.Host == "" {
	return errors.New("Domain should not be blank")
}
```
Link to [playground](https://play.golang.org/p/5WwZifw_ikQ).